modules/swagger-codegen/src/main/resources/erlang-server/types.mustache

```diff
@@ -9,11 +9,23 @@
 -export_type([operation_id/0]).
 -export_type([api_key/0]).
 -export_type([object/0]).
-
--type auth_context() :: any().
--type operation_id() :: atom().
--type api_key()      :: binary().
--type object()       :: map().
+-export_type([logic_handler/1]).
+-export_type([handler_opts/1]).
+-export_type([status/0]).
+-export_type([headers/0]).
+-export_type([response_body/0]).
+-export_type([response/0]).
+
+-type auth_context()    :: any().
+-type operation_id()    :: atom().
+-type api_key()         :: binary().
+-type object()          :: map().
+-type handler_opts(T)   :: {packageName}}_logic_handler:handler_opts(T).
+-type logic_handler(T)  :: {packageName}}_logic_handler:handler(T).
```

Comment:
Btw, this look like cyclic dependency, since `{{packageName}}_logic_handler` imports some types from here.

I believe it's ok to import `{{packageName}}_logic_handler:handler()` type where you will need it without the requirement to reexport him here first.

Reply:
The reexports in this module() are generally done for the library interface types, so a user code won't have to contain generated long (and usually ugly) module names with original type specs.

Yes, sometimes this leads to cyclic dependancies for types.

Reply:
I personally consider cyclic dependencies as being _ugly_, a lot more ugly than some long module name (which I will type in as a name of behaviour anyway).

> The reexports in this module are generally done for the library interface types

Well, that's still possible without incurring cyclic dependencies.